### DISCUSSION TITLE
DFPlayer_Mini на чипе MH2024K-24SS,  Ошибка стека!

### DISCUSSION THREAD

**Discussion by Uef1** [vvip-68/GyverPanelWiFi#235] [Задавайте вопросы]
Доброго дня,

Не удаётся запустить DFPlayer_Mini на чипе MH2024K-24SS, нормально работает с библиотекой DFRobotDFPlayerMini, но не хочет с DFMiniMp3 от Makuha. Библтотека установлена из гитхаба. 
Не понимаю почему в плеер отправляется команда "4E" , в  описании вообще такой команды нет.

![image](https://user-images.githubusercontent.com/107195919/172863459-5e27d55b-be27-4e14-ba7e-c188ed808bc1.png)

Пните в нужном направлении.

**Comment by vvip-68** [comment]
Команда 4E отправляется, потому что при инициализации плеера в скетче выполняется запрос получения количества файлов, находящихся в папке со звуками. Это примерно строка 56 a скетче в файле dfplayer.ino

![image](https://user-images.githubusercontent.com/39353139/172910493-d9bd2d07-dc67-4307-a0ad-6bf30f531ba4.png)

Далее строки 62, 73, 76, 87 и 90 - идет примерно тот же запрос.   

Это вызов функции dfPlayer.getFolderTrackCount(N);

Эта функция описана в библиотеке от Makuha, файл DFMiniMp3.h

![image](https://user-images.githubusercontent.com/39353139/172911013-d9acb2ee-90e4-4573-ab8f-4879f4a3900e.png)

Как видим - код команды получения количества файлов в папке - 0x4E
Об этом свидетельствует спецификация (datasheet) на плеер - 

![image](https://user-images.githubusercontent.com/39353139/172911313-6fc51941-680f-4925-8de4-e3d6a74e662f.png)

или, согласно примерам из той же документации - 

![image](https://user-images.githubusercontent.com/39353139/172911641-9c7f1a40-8b58-4214-b93b-c8bebdfbb32c.png)

**Comment by vvip-68** [comment]
[FN-M16P Embedded MP3 Audio Module Datasheet.pdf](https://github.com/vvip-68/GyverPanelWiFi/files/8872723/FN-M16P.Embedded.MP3.Audio.Module.Datasheet.pdf)

**Comment by vvip-68** [comment]
Есть еще варианты даташита не плеер, например - с такой таблицей

![image](https://user-images.githubusercontent.com/39353139/172913654-83188c64-82b8-4dc5-b88c-e26d698acb27.png)

Здесь команда 0x4E не описана, да и вообще отсутствует команда получения количества файлов из указанной папки, только количество файлов в целом - вроде 0x47

Попробуйте посмотреть ту библиотеку с которой у вас плеер работает - какой запрос на получение кол-ва файлов в указанной папке выдается там (как реализована команда dfPlayer.getFolderTrackCount(N)) . И исправить в библиотеке от Makuha код команды на тот, с которым работает.

**Comment by vvip-68** [comment]
Впрочем, в библиотеке DFRobotDFPlayerMini (файл DFRobotDFPlayerMini.cpp) эта функция (получения кол-ва файлов в папке) описана так:

![image](https://user-images.githubusercontent.com/39353139/172915040-86e8be23-671c-4008-98ab-db0004854958.png)

Как видим - отправляется все тот же код команды 0x4E

**Comment by Uef1** [comment]
Добрый день,
Спасибо, за подробный ответ.
Хорошего дня!

**Comment by vvip-68** [comment]
Какое напряжение питания DFPlayer?
Вот чисто для эксперимента - отключи от него динамик, и запитай его от не более чем 4.2 вольта. Это максимум по даташиту.
Часто при превышении питающего напряжения он начинает "чудить". Например перестает распознавать команды, сыпать ошибками...

**Reply by Uef1** [reply]
Питание на плеере 4.5V,

**Comment by Uef1** [comment]
Питание на плеере 4.5V, Похоже какая-то проблема с портом/портами. 
Плеер начал работать после того как изменил порты:
#define STX D5                  // подключен в RX пин (2) модуля DFPlayer 
#define SRX D6                  // подключен в TX пин (3) модуля DFPlayer
Установил режим без контрольных сумм:
#define DFPLAYER_TYPE 1
С этими установками все равно один раз выдает ошибку стека, но файлы находит и проигрывает.
![image](https://user-images.githubusercontent.com/107195919/173603208-f3697ee1-2e85-41b8-aad9-dc0dbd3dadbe.png)

Спасибо.
